File: Python/01.ProgrammingBasics/99.Pre-Exam/04.Workout/_04.Workout.py
import math

days = int(input())
initial_distance = float(input())
total_distance = initial_distance

for _ in range(days):
    increase_percentage = int(input())
    initial_distance += (increase_percentage / 100.0) * initial_distance
    total_distance += initial_distance

extra_distance = math.ceil(abs(total_distance - 1000))

if total_distance >= 1000:
    print(f"You've done a great job running {int(extra_distance)} more kilometers!")
else:
    print(f"Sorry Mrs. Ivanova, you need to run {int(extra_distance)} more kilometers")

File: JS/01.Basics/99.Pre-Exam/04.Workout.js
function training(input) {
    let days = Number(input.shift());
    let initialKilometers = Number(input.shift());
    let totalKilometers = initialKilometers;

    for (let i = 0; i < days; i++) {
        let percentageIncrease = Number(input.shift());
        initialKilometers += (percentageIncrease / 100) * initialKilometers;
        totalKilometers += initialKilometers;
    }

    if (totalKilometers >= 1000) {
        console.log(`You've done a great job running ${Math.floor(totalKilometers - 1000 + 1)} more kilometers!`);
    } else {
        console.log(`Sorry Mrs. Ivanova, you need to run ${Math.ceil(1000 - totalKilometers)} more kilometers`);
    }
}

training(["5", "30", "10", "15", "20", "5", "12"]);
training(["4", "100", "30", "50", "60", "80"]);
